### DISCUSSION TITLE
GITHUB_TOKEN not accessable from fork, how to replace

### DISCUSSION THREAD

**Discussion by Poolitzer** [community/community#25358] [Actions]
<p>Hello guys.</p>
<p>I created a very simple <a href="https://github.com/Poolitzer/notifier-action" rel="nofollow noopener">notifier</a>, which uses the GITHUB_TOKEN variable to send a message in the PR concersation whenever a file in a folder changes. This works perfectly fine from PR from different branches, but fails when the PR originates from a fork with the error_Resource not accessible by integration_. <a href="https://github.com/python-telegram-bot/python-telegram-bot/pull/1622/checks?check_run_id=309159621#step:2:5" rel="nofollow noopener">example.</a></p>
<p>Now I did some research. <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets#using-encrypted-secrets-in-a-workflow" rel="nofollow noopener">This article</a> states that I can access the token from a fork, but <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token#permissions-for-the-github_token" rel="nofollow noopener">this one</a> says that the token only has read permission when originating from a fork. So I am going to assume that I need issue write permission to create a message in the PR conversation, right?</p>
<p>If that is correct, my question is how to proceed. Do I need to create a Github App? It sounds appropriate, but it looks a bit complicated and I would love to see a better solution.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/poolitzer">@poolitzer</a> ,</p>
<p>As github stuff replied <a href="https://github.community/t/optional-read-write-permission-for-forked-repos/16330/2">here</a>, it’s not supported/planed to give forks the ablity to write to main repo directly. I assume there’s no easy way to comment the pull request from forked repo, thanks.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/6b605bec97aa2b71e32c26d1ff6f37e66e54669b.png" data-download-href="/uploads/short-url/fjTxkXUSqMdgzJYKl985cTgjauD.png?dl=1" title="secrets.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181090602-298be032-f398-4650-b52a-5c34e05abf1d.png" alt="secrets.png" data-base62-sha1="fjTxkXUSqMdgzJYKl985cTgjauD" width="690" height="142" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/6b605bec97aa2b71e32c26d1ff6f37e66e54669b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">secrets.png</span><span class="informations">748×155 6.21 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Poolitzer** [comment]
<p>Hey <a class="mention" href="/weide-zhou">@weide-zhou</a>,</p>
<p>thanks for that, I know now that I can’t use actions directly then, great.</p>
